[♻️Refactor] 화면 너비값 관리 (resize event)

dkslel1225 opened the pull request:
<!-- [깃모지 타입] -->
<!--  ✨Feat  🐛Fix  🎨Style  🗑️Remove  ♻️Refactor  📝Docs  🚀Deploy  -->
<!--  📁Chore : 폴더 구조 변경 또는 디렉토리 작업  🔧Chore : 설정, 빌드 변경  -->

## ✨ 요약
* isDesktop, isTablet, isMobile을 useWindowSize에서 관리하도록 수정했습니다.(전역상태 따로 사용 X)
  값의 타입은 boolean입니다.

<!-- 구현 내용or변경 사항에 대한 간단한 설명 -->
<!-- 관련 이슈는 development로 연결 또는 직접 작성( #issue, closes #issue ) -->

## 📝 상세 내용
<!-- 구현 내용에 대한 자세한 설명 -->
### 사용 방법
* 만약, **SSR환경인 컴포넌트인데 + isDesktop등을 체크하는 로직이 들어있다면**, ready 상태값을 가져가서 사용해 주세요. 
  * ready: SSR일때의 하이드레이션 미스매치 에러를 방지하기 위해 추가한 상태값.
  * window가 undefined가 아니고, 화면 사이즈에 대한 상태값이 업데이트되었을 때 true로 변경됨
  * ```jsx
    // ready 사용 예시
    const { isDesktop, isTablet, isMobile, ready } = useWindowSize();
    if(!ready) return;
    ```
* CSR 컴포넌트면 - ready 사용 필요X
  ```jsx
    const { isDesktop, isTablet, isMobile } = useWindowSize();
  ```

## 🖼️ 스크린샷

<!-- UI 변경이 있는 경우 변경 전/후 스크린샷 -->

## ✅ 체크리스트

- [x] 브랜치 네이밍 컨벤션을 준수했습니다
- [x] 커밋 컨벤션을 준수했습니다
- [x] 코드가 프로젝트의 스타일 가이드라인을 준수합니다

## 💡 참고 사항
* useModalStore에서 isDesktop 제거할 예정입니다.
   * ReserveCalendarPage에서 사용중인 isDesktop,
   * src/app/my/layout.tsx에 있는 isMobile 을
   --> useWindowSize()에서 가져오는거로 수정해주세용

+ ReserveCalendarPage는 제가 다음 PR에서 일부 수정해두긴 했는데, 굳이 isDesktop체크 안해도 될 것 같아요. 일단은 준우님꺼 합치는 과정에서 conflict생기면 혼란스러울것 같아서 냅둠

* setTimeout, debounce 등을 활용한 성능 최적화는, 적용했을 때 체험 상세페이지의 ui 변환이 어색해져서 일단 보류했습니다
(화면 리사이즈 종료 후 0.3초 이후에 화면상태 값을 업데이트하면, 그만큼 반응도 느리고 lg:,md:등의 스타일은 먼저 적용되어버려서 ui가 이상한 구간이 생김.. 나중에 상세페이지 리팩토링하면서 최적화 적용 시도해 볼게요)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added richer modal interactions and a demo page to preview standard and adaptive modals.
* **Bug Fixes**
  * Fixed inconsistent open/close behavior and login/confirmation flows across screen sizes.
  * Corrected tablet detection to prevent layout glitches.
* **Refactor**
  * Unified device-size handling for more consistent responsive behavior and smoother modal transitions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
♻️ refactor: useWindowSize.ts에 화면 사이즈 상태값 추가 및 관리

0Hyunn reviewed (approved):
확인했습니다! 윈도우 사이즈별 처리가 되어있네요!!

Commit:
🐰refactor: SSR환경에서의 React hydration mismatch 방지하기 위해 ready 상태값 추가힘
- isDesktop등을 반영하는 스타일이 있는 경우 + SSR방식으로 렌더링 되는 경우 -> ready여부 체크하여 렌더링 에러 방지하는걸 권장함
